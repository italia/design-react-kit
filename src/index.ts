export {
  BreadcrumbItem,
  ButtonGroup,
  ButtonToolbar,
  CardColumns,
  CardDeck,
  CardFooter,
  CardGroup,
  CardHeader,
  CardImg,
  CardImgOverlay,
  CardLink,
  CardSubtitle,
  Col,
  Container,
  Fade,
  Form,
  FormFeedback,
  FormGroup,
  FormText,
  InputGroup,
  InputGroupText,
  Label,
  ListGroup,
  ListGroupItem,
  ListGroupItemHeading,
  ListGroupItemText,
  Media,
  NavItem,
  NavLink,
  Navbar,
  NavbarBrand,
  NavbarToggler,
  PaginationItem as PagerItem,
  PaginationLink as PagerLink,
  Popover,
  PopoverBody,
  PopoverHeader,
  Row,
  TabContent,
  TabPane,
  Table,
  Tooltip,
  UncontrolledAlert,
  UncontrolledCollapse,
  UncontrolledTooltip,
  Util
} from 'reactstrap';

export { Accordion } from './Accordion/Accordion';
export { AccordionBody } from './Accordion/AccordionBody';
export { AccordionHeader } from './Accordion/AccordionHeader';
export { AccordionItem } from './Accordion/AccordionItem';
export { Alert } from './Alert/Alert';
export { AvatarContainer } from './Avatar/AvatarContainer';
export { AvatarExtraText } from './Avatar/AvatarExtraText';
export { AvatarGroupContainer } from './Avatar/AvatarGroupContainer';
export { AvatarIcon } from './Avatar/AvatarIcon';
export { AvatarPresence } from './Avatar/AvatarPresence';
export { AvatarStatus } from './Avatar/AvatarStatus';
export { AvatarWrapper } from './Avatar/AvatarWrapper';
export { BackToTop } from './BackToTop/BackToTop';
export { Badge } from './Badge/Badge';
export { BottomNav } from './BottomNav/BottomNav';
export { BottomNavItem } from './BottomNav/BottomNavItem';
export { Breadcrumb } from './Breadcrumb/Breadcrumb';
export { Button } from './Button/Button';
export { Callout } from './Callout/Callout';
export { CalloutMoreFooter } from './Callout/CalloutMoreFooter';
export { CalloutText } from './Callout/CalloutText';
export { CalloutTitle } from './Callout/CalloutTitle';
export { Card } from './Card/Card';
export { CardBody } from './Card/CardBody';
export { CardCategory } from './Card/CardCategory';
export { CardFooterCTA } from './Card/CardFooterCTA';
export { CardReadMore } from './Card/CardReadMore';
export { CardSignature } from './Card/CardSignature';
export { CardTag } from './Card/CardTag';
export { CardTagsHeader } from './Card/CardTagsHeader';
export { CardText } from './Card/CardText';
export { CardTitle } from './Card/CardTitle';
export { Chip } from './Chips/Chip';
export { ChipLabel } from './Chips/ChipLabel';
export { Collapse } from './Collapse/Collapse';
export { Dimmer } from './Dimmer/Dimmer';
export { DimmerButtons } from './Dimmer/DimmerButtons';
export { Dropdown } from './Dropdown/Dropdown';
export { DropdownMenu } from './Dropdown/DropdownMenu';
export { DropdownToggle } from './Dropdown/DropdownToggle';
export { FontLoader, useFontLoader } from './FontLoader/FontLoader';
export { Forward } from './Forward/Forward';
export { GoBack } from './GoBack/GoBack';
export { GridItem, GridItemText, GridItemTextWrapper, GridList, GridRow } from './Grid/index';
export { Header } from './Header/Header';
export { HeaderBrand } from './Header/HeaderBrand';
export { HeaderContent } from './Header/HeaderContent';
export { HeaderLinkZone } from './Header/HeaderLinkZone';
export { HeaderRightZone } from './Header/HeaderRightZone';
export { HeaderSearch } from './Header/HeaderSearch';
export { HeaderSocialsZone } from './Header/HeaderSocialsZone';
export { HeaderToggler } from './Header/HeaderToggler';
export { Headers } from './Header/Headers';
export { Hero, HeroBackground, HeroBody, HeroButton, HeroCategory, HeroTitle } from './Hero/index';
export { Icon, clearIconCache, iconsList as icons, preloadIcons } from './Icon/Icon';
export { Input } from './Input/Input';
export { InputContainer } from './Input/InputContainer';
export { TextArea } from './Input/TextArea';
export { LinkList } from './LinkList/LinkList';
export { LinkListItem } from './LinkList/LinkListItem';
export { MegamenuFooter } from './Megamenu/MegamenuFooter';
export { MegamenuHighlightColumn } from './Megamenu/MegamenuHighlightColumn';
export { MegamenuItem } from './Megamenu/MegamenuItem';
export { Modal } from './Modal/Modal';
export { ModalBody } from './Modal/ModalBody';
export { ModalFooter } from './Modal/ModalFooter';
export { ModalHeader } from './Modal/ModalHeader';
export { Nav } from './Nav/Nav';
export { useNavScroll } from './NavScroll/index';
export { NotificationManager, createNotification, notify } from './Notification/index';
export { Pager } from './Pager/Pager';
export { Progress } from './Progress/Progress';
export { Rating } from './Rating/Rating';
export { ResponsiveImage } from './ResponsiveImage/ResponsiveImage';
export { Section } from './Section/Section';
export { Select } from './Select/Select';
export { Sidebar } from './Sidebar/Sidebar';
export { Skiplink } from './Skiplink/Skiplink';
export { SkiplinkItem } from './Skiplink/SkiplinkItem';
export { Spinner } from './Spinner/Spinner';
export { StepperContainer } from './Stepper/StepperContainer';
export { StepperContent } from './Stepper/StepperContent';
export { StepperDots } from './Stepper/StepperDots';
export { StepperHeader } from './Stepper/StepperHeader';
export { StepperHeaderElement } from './Stepper/StepperHeaderElement';
export { StepperNav } from './Stepper/StepperNav';
export { ThumbNav } from './ThumbNav/ThumbNav';
export { ThumbNavItem } from './ThumbNav/ThumbNavItem';
export { TimelinePin } from './Timeline/TimelinePin';
export { Timeline } from './Timeline/TimelineWrapper';
export { Toggle } from './Toggle/Toggle';
export { Toolbar } from './Toolbar/Toolbar';
export { ToolbarDividerItem } from './Toolbar/ToolbarDividerItem';
export { ToolbarItem } from './Toolbar/ToolbarItem';

// Types
export type { AccordionProps } from './Accordion/Accordion';
export type { AccordionBodyProps } from './Accordion/AccordionBody';
export type { AccordionHeaderProps } from './Accordion/AccordionHeader';
export type { AlertProps } from './Alert/Alert';
export type { AvatarContainerProps } from './Avatar/AvatarContainer';
export type { AvatarExtraTextProps } from './Avatar/AvatarExtraText';
export type { AvatarGroupContainerProps } from './Avatar/AvatarGroupContainer';
export type { AvatarIconProps } from './Avatar/AvatarIcon';
export type { AvatarPresenceProps } from './Avatar/AvatarPresence';
export type { AvatarStatusProps } from './Avatar/AvatarStatus';
export type { AvatarWrapperProps } from './Avatar/AvatarWrapper';
export type { BadgeProps } from './Badge/Badge';
export type { BottomNavProps } from './BottomNav/BottomNav';
export type { BottomNavItemProps } from './BottomNav/BottomNavItem';
export type { BreadcrumbProps } from './Breadcrumb/Breadcrumb';
export type { ButtonProps } from './Button/Button';
export type { CalloutProps } from './Callout/Callout';
export type { CalloutTextProps } from './Callout/CalloutText';
export type { CalloutTitleProps } from './Callout/CalloutTitle';
export type { CardProps } from './Card/Card';
export type { CardBodyProps } from './Card/CardBody';
export type { CardCategoryProps } from './Card/CardCategory';
export type { CardFooterCTAProps } from './Card/CardFooterCTA';
export type { CardReadMoreProps } from './Card/CardReadMore';
export type { CardSignatureProps } from './Card/CardSignature';
export type { CardTagProps } from './Card/CardTag';
export type { CardTagsHeaderProps } from './Card/CardTagsHeader';
export type { CardTextProps } from './Card/CardText';
export type { CardTitleProps } from './Card/CardTitle';
export type { ChipProps } from './Chips/Chip';
export type { ChipLabelProps } from './Chips/ChipLabel';
export type { DimmerProps } from './Dimmer/Dimmer';
export type { DimmerButtonsProps } from './Dimmer/DimmerButtons';
export type { DropdownProps } from './Dropdown/Dropdown';
export type { DropdownMenuProps } from './Dropdown/DropdownMenu';
export type { DropdownToggleProps } from './Dropdown/DropdownToggle';
export type { FontLoaderProps } from './FontLoader/FontLoader';
export type { ForwardProps } from './Forward/Forward';
export type {
  GridItemProps,
  GridItemTextProps,
  GridItemTextWrapperProps,
  GridListProps,
  GridRowProps
} from './Grid/index';
export type { HeaderProps } from './Header/Header';
export type { HeaderBrandProps } from './Header/HeaderBrand';
export type { HeaderContentProps } from './Header/HeaderContent';
export type { HeaderLinkZoneProps } from './Header/HeaderLinkZone';
export type { HeaderRightZoneProps } from './Header/HeaderRightZone';
export type { HeaderSearchProps } from './Header/HeaderSearch';
export type { HeaderSocialsZoneProps } from './Header/HeaderSocialsZone';
export type { HeaderTogglerProps } from './Header/HeaderToggler';
export type { HeadersProps } from './Header/Headers';
export type {
  HeroBackgroundProps,
  HeroBodyProps,
  HeroButtonProps,
  HeroCategoryProps,
  HeroProps,
  HeroTitleProps
} from './Hero/index';
export type { IconName, IconProps } from './Icon/Icon';
export type { InputProps } from './Input/Input';
export type { InputContainerProps } from './Input/InputContainer';
export type { TextAreaProps } from './Input/TextArea';
export type { LinkListProps } from './LinkList/LinkList';
export type { LinkListItemProps } from './LinkList/LinkListItem';
export type { MegamenuFooterProps } from './Megamenu/MegamenuFooter';
export type { MegamenuHCProps } from './Megamenu/MegamenuHighlightColumn';
export type { MegamenuItemProps } from './Megamenu/MegamenuItem';
export type { ModalProps } from './Modal/Modal';
export type { ModalBodyProps } from './Modal/ModalBody';
export type { ModalFooterProps } from './Modal/ModalFooter';
export type { ModalHeaderProps } from './Modal/ModalHeader';
export type { NavProps } from './Nav/Nav';
export type { ChangesType, RegisterOptions, useNavScrollArgs, useNavScrollResult } from './NavScroll';
export type { NotificationId, NotificationManagerProps, NotificationOptions, NotificationProps } from './Notification';
export type { PagerProps } from './Pager/Pager';
export type { ProgressProps } from './Progress/Progress';
export type { RatingProps } from './Rating/Rating';
export type { ResponsiveImageProps } from './ResponsiveImage/ResponsiveImage';
export type { SectionProps } from './Section/Section';
export type { SelectProps } from './Select/Select';
export type { SidebarProps } from './Sidebar/Sidebar';
export type { SkiplinkProps } from './Skiplink/Skiplink';
export type { SkiplinkItemProps } from './Skiplink/SkiplinkItem';
export type { SpinnerProps } from './Spinner/Spinner';
export type { StepperContainerProps } from './Stepper/StepperContainer';
export type { StepperContentProps } from './Stepper/StepperContent';
export type { StepperDotsProps } from './Stepper/StepperDots';
export type { StepperHeaderProps } from './Stepper/StepperHeader';
export type { StepperHeaderElementProps } from './Stepper/StepperHeaderElement';
export type { StepperNavProps } from './Stepper/StepperNav';
export type { ThumbNavProps } from './ThumbNav/ThumbNav';
export type { ThumbNavItemProps } from './ThumbNav/ThumbNavItem';
export type { TimelinePinProps } from './Timeline/TimelinePin';
export type { TimelineProps } from './Timeline/TimelineWrapper';
export type { ToggleProps } from './Toggle/Toggle';
export type { ToolbarProps } from './Toolbar/Toolbar';
export type { ToolbarItemBadge, ToolbarItemProps } from './Toolbar/ToolbarItem';

export type {
  BreadcrumbItemProps,
  ButtonDropdownProps,
  ButtonGroupProps,
  ButtonToolbarProps,
  CardColumnsProps,
  CardDeckProps,
  CardFooterProps,
  CardGroupProps,
  CardHeaderProps,
  CardImgOverlayProps,
  CardImgProps,
  CardLinkProps,
  CardSubtitleProps,
  ColProps,
  ContainerProps,
  FadeProps,
  FormFeedbackProps,
  FormGroupProps,
  FormProps,
  FormTextProps,
  InputGroupProps,
  InputGroupTextProps,
  LabelProps,
  ListGroupItemHeadingProps,
  ListGroupItemProps,
  ListGroupItemTextProps,
  ListGroupProps,
  MediaProps,
  NavItemProps,
  NavLinkProps,
  NavbarBrandProps,
  NavbarProps,
  NavbarTogglerProps,
  PaginationItem as PagerItemProps,
  PaginationLink as PagerLinkProps,
  PopoverBodyProps,
  PopoverHeaderProps,
  PopoverProps,
  RowProps,
  TabContentProps,
  TabPaneProps,
  TableProps,
  TooltipProps,
  UncontrolledAlertProps,
  UncontrolledCollapseProps,
  UncontrolledTooltipProps
} from 'reactstrap';
