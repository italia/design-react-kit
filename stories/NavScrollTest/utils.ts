export function getStorybookExtraStyles() {
  return insideIframe() && insideDocsPage()
    ? {
        maxHeight: 600,
        overflowY: 'scroll' as const
      }
    : {};
}

function insideDocsPage() {
  return document.querySelector('.innerZoomElementWrapper');
}

function insideIframe() {
  try {
    return window.self !== window.top;
  } catch (e) {
    return true;
  }
}
