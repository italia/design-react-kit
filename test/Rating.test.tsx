import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import * as Utils from '../src/utils';
import { isCustomLegendObject } from '../src/Rating/Rating';
import { Rating, preloadIcons } from '../src';

jest.mock('./../src/utils', () => {
  const utilModule = jest.requireActual('./../src/utils');
  return { ...utilModule, noop: jest.fn() };
});

describe('Rating', () => {
  // Icons are now async, so preload them to make it behave in an sync way
  beforeAll(() => preloadIcons(['it-star-full']));
  it('Should not apply className to the wrapper', () => {
    const { container } = render(
      <Rating
        value={4}
        inputs={['star1c', 'star2c', 'star3c', 'star4c', 'star5c']}
        name='rating'
        className={'myClass'}
      />
    );
    expect(container.firstChild).not.toHaveClass('myClass');
  });

  it('Should apply wrapperClassName to the wrapper', () => {
    const { container } = render(
      <Rating
        value={4}
        inputs={['star1c', 'star2c', 'star3c', 'star4c', 'star5c']}
        name='rating'
        wrapperClassName={'myClass'}
      />
    );
    expect(container.firstChild).toHaveClass('myClass');
  });

  describe('Read only', () => {
    it('Should not apply readOnly className to the wrapper by default', () => {
      const { container } = render(
        <Rating
          value={4}
          inputs={['star1c', 'star2c', 'star3c', 'star4c', 'star5c']}
          name='rating'
          wrapperClassName={'myClass'}
        />
      );
      expect(container.firstChild).not.toHaveClass('rating-read-only');
    });

    it('Should apply readOnly className to the wrapper when readOnly', () => {
      const { container } = render(
        <Rating
          value={4}
          inputs={['star1c', 'star2c', 'star3c', 'star4c', 'star5c']}
          name='rating'
          wrapperClassName={'myClass'}
          readOnly
        />
      );
      expect(container.firstChild).toHaveClass('rating-read-only');
    });

    it('Should not apply set aria-hidden by default', () => {
      const { container } = render(
        <Rating value={4} inputs={['star1c', 'star2c', 'star3c', 'star4c', 'star5c']} name='rating' />
      );
      const ariaHiddenInputs = container.querySelectorAll('input[aria-hidden="true"]');
      expect(ariaHiddenInputs).toHaveLength(0);
    });

    it('Should set aria-hidden when readOnly', () => {
      const { container } = render(
        <Rating value={4} inputs={['star1c', 'star2c', 'star3c', 'star4c', 'star5c']} name='rating' readOnly />
      );
      const ariaHiddenInputs = container.querySelectorAll('input[aria-hidden="true"]');
      expect(ariaHiddenInputs).toHaveLength(5);
    });

    it('Should use onChangeRating props by default', () => {
      const onChangeRating = jest.fn();
      const { container } = render(
        <Rating
          value={4}
          inputs={['star1c', 'star2c', 'star3c', 'star4c', 'star5c']}
          name='rating'
          onChangeRating={onChangeRating}
        />
      );
      const input = container.querySelectorAll('input')[0];
      fireEvent.click(input);
      expect(onChangeRating).toHaveBeenCalledWith(5, 'rating');
      expect(onChangeRating).toHaveBeenCalledTimes(1);
    });

    it('Should use noop props when readOnly', () => {
      const { container } = render(
        <Rating value={4} inputs={['star1c', 'star2c', 'star3c', 'star4c', 'star5c']} name='rating' />
      );
      const input = container.querySelectorAll('input')[0];
      fireEvent.click(input);
      expect(Utils.noop).toHaveBeenCalledTimes(1);
    });

    it('Should not make input disabled by default', () => {
      const { container } = render(
        <Rating value={4} inputs={['star1c', 'star2c', 'star3c', 'star4c', 'star5c']} name='rating' />
      );
      const inputs = container.querySelectorAll('input');
      inputs.forEach((input) => expect(input.disabled).toBeFalsy());
    });

    it('Should make input disabled when readOnly', () => {
      const { container } = render(
        <Rating value={4} inputs={['star1c', 'star2c', 'star3c', 'star4c', 'star5c']} name='rating' readOnly />
      );
      const inputs = container.querySelectorAll('input');
      inputs.forEach((input) => expect(input.disabled).toBeTruthy());
    });
  });

  describe('isCustomLegendObject', () => {
    it('Should return true', () => {
      expect(isCustomLegendObject({ content: true, srOnly: true })).toBeTruthy();
    });

    it('Should return false', () => {
      expect(isCustomLegendObject(null)).toBeFalsy();
      expect(isCustomLegendObject('null')).toBeFalsy();
      expect(isCustomLegendObject(<></>)).toBeFalsy();
    });
  });

  describe('Legend', () => {
    it('Should render <legend> tag when legend is a string', () => {
      const { container } = render(
        <Rating
          value={4}
          inputs={['star1c', 'star2c', 'star3c', 'star4c', 'star5c']}
          name='rating'
          legend='This is a legend'
        />
      );
      const legend = container.querySelectorAll('legend');
      expect(legend).toHaveLength(1);
    });

    it('Should render <legend> tag when using a custom legend', () => {
      const { container } = render(
        <Rating
          value={4}
          inputs={['star1c', 'star2c', 'star3c', 'star4c', 'star5c']}
          name='rating'
          legend={{ content: 'This is a legend', srOnly: false }}
        />
      );
      const legend = container.querySelectorAll('legend');
      expect(legend).toHaveLength(1);
    });

    it('Should render legend as is', () => {
      const { container } = render(
        <Rating
          value={4}
          inputs={['star1c', 'star2c', 'star3c', 'star4c', 'star5c']}
          name='rating'
          legend={<div id='legend' />}
        />
      );
      const legend = container.querySelectorAll('#legend');
      expect(legend).toHaveLength(1);
    });
  });

  describe('Label Template', () => {
    it('Should use default label template', () => {
      render(<Rating value={4} inputs={['star1c', 'star2c', 'star3c', 'star4c', 'star5c']} name='rating' />);
      const legend = screen.getByText('Valuta 1 stelle su 5');
      expect(legend).not.toBe(null);
    });

    it('Should use customer label template', () => {
      render(
        <Rating
          value={4}
          inputs={['star1c', 'star2c', 'star3c', 'star4c', 'star5c']}
          name='rating'
          labelTemplate={(vote) => `Vote ${vote}/5`}
        />
      );
      const legend = screen.getByText('Vote 1/5');
      expect(legend).not.toBe(null);
    });
  });
});

test('should have a testId for resilient UI changes', () => {
  render(<Rating inputs={['star1c', 'star2c', 'star3c', 'star4c', 'star5c']} name='rating' testId='test-id-rating' />);

  expect(screen.getByTestId('test-id-rating')).toBeTruthy();
});
