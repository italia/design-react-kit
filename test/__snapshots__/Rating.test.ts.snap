// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stories Snapshots Documentazione/Componenti/Rating RatingBase 1`] = `
<div>
  <fieldset
    class="rating rating-label mb-3"
  >
    <legend
      class=""
    >
      Rating
    </legend>
    <input
      aria-describedby="star5aDescription"
      class=""
      id="star5a"
      name="ratingA"
      type="radio"
      value="5"
    />
    <label
      class="full form-label"
      for="star5a"
    >
      <svg
        class="icon icon-sm"
        enable-background="new 0 0 24 24"
        role="img"
        viewBox="0 0 24 24"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          icon
        </title>
        <path
          d="M12 1.7 9.5 9.2H1.6L8 13.9l-2.4 7.6 6.4-4.7 6.4 4.7-2.4-7.6 6.4-4.7h-7.9L12 1.7z"
        />
      </svg>
      <span
        class="visually-hidden"
      >
        Valuta 5 stelle su 5
      </span>
    </label>
    <input
      aria-describedby="star4aDescription"
      class=""
      id="star4a"
      name="ratingA"
      type="radio"
      value="4"
    />
    <label
      class="full form-label"
      for="star4a"
    >
      <svg
        class="icon icon-sm"
        enable-background="new 0 0 24 24"
        role="img"
        viewBox="0 0 24 24"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          icon
        </title>
        <path
          d="M12 1.7 9.5 9.2H1.6L8 13.9l-2.4 7.6 6.4-4.7 6.4 4.7-2.4-7.6 6.4-4.7h-7.9L12 1.7z"
        />
      </svg>
      <span
        class="visually-hidden"
      >
        Valuta 4 stelle su 5
      </span>
    </label>
    <input
      aria-describedby="star3aDescription"
      class=""
      id="star3a"
      name="ratingA"
      type="radio"
      value="3"
    />
    <label
      class="full form-label"
      for="star3a"
    >
      <svg
        class="icon icon-sm"
        enable-background="new 0 0 24 24"
        role="img"
        viewBox="0 0 24 24"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          icon
        </title>
        <path
          d="M12 1.7 9.5 9.2H1.6L8 13.9l-2.4 7.6 6.4-4.7 6.4 4.7-2.4-7.6 6.4-4.7h-7.9L12 1.7z"
        />
      </svg>
      <span
        class="visually-hidden"
      >
        Valuta 3 stelle su 5
      </span>
    </label>
    <input
      aria-describedby="star2aDescription"
      class=""
      id="star2a"
      name="ratingA"
      type="radio"
      value="2"
    />
    <label
      class="full form-label"
      for="star2a"
    >
      <svg
        class="icon icon-sm"
        enable-background="new 0 0 24 24"
        role="img"
        viewBox="0 0 24 24"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          icon
        </title>
        <path
          d="M12 1.7 9.5 9.2H1.6L8 13.9l-2.4 7.6 6.4-4.7 6.4 4.7-2.4-7.6 6.4-4.7h-7.9L12 1.7z"
        />
      </svg>
      <span
        class="visually-hidden"
      >
        Valuta 2 stelle su 5
      </span>
    </label>
    <input
      aria-describedby="star1aDescription"
      class=""
      id="star1a"
      name="ratingA"
      type="radio"
      value="1"
    />
    <label
      class="full form-label"
      for="star1a"
    >
      <svg
        class="icon icon-sm"
        enable-background="new 0 0 24 24"
        role="img"
        viewBox="0 0 24 24"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          icon
        </title>
        <path
          d="M12 1.7 9.5 9.2H1.6L8 13.9l-2.4 7.6 6.4-4.7 6.4 4.7-2.4-7.6 6.4-4.7h-7.9L12 1.7z"
        />
      </svg>
      <span
        class="visually-hidden"
      >
        Valuta 1 stelle su 5
      </span>
    </label>
  </fieldset>
</div>
`;

exports[`Stories Snapshots Documentazione/Componenti/Rating RatingConLabel 1`] = `
<div>
  <fieldset
    class="rating rating-label mb-3"
  >
    <span
      class="visually-hidden"
    >
      Valutazione
    </span>
     
    <span>
      4
       stelle
    </span>
    <span
      class="visually-hidden"
    >
      su 5
    </span>
    <input
      aria-describedby="star5bDescription"
      class=""
      id="star5b"
      name="ratingB"
      type="radio"
      value="5"
    />
    <label
      class="full form-label"
      for="star5b"
    >
      <svg
        class="icon icon-sm"
        enable-background="new 0 0 24 24"
        role="img"
        viewBox="0 0 24 24"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          icon
        </title>
        <path
          d="M12 1.7 9.5 9.2H1.6L8 13.9l-2.4 7.6 6.4-4.7 6.4 4.7-2.4-7.6 6.4-4.7h-7.9L12 1.7z"
        />
      </svg>
      <span
        class="visually-hidden"
      >
        Valuta 5 stelle su 5
      </span>
    </label>
    <input
      aria-describedby="star4bDescription"
      checked=""
      class=""
      id="star4b"
      name="ratingB"
      type="radio"
      value="4"
    />
    <label
      class="full form-label"
      for="star4b"
    >
      <svg
        class="icon icon-sm"
        enable-background="new 0 0 24 24"
        role="img"
        viewBox="0 0 24 24"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          icon
        </title>
        <path
          d="M12 1.7 9.5 9.2H1.6L8 13.9l-2.4 7.6 6.4-4.7 6.4 4.7-2.4-7.6 6.4-4.7h-7.9L12 1.7z"
        />
      </svg>
      <span
        class="visually-hidden"
      >
        Valuta 4 stelle su 5
      </span>
    </label>
    <input
      aria-describedby="star3bDescription"
      class=""
      id="star3b"
      name="ratingB"
      type="radio"
      value="3"
    />
    <label
      class="full form-label"
      for="star3b"
    >
      <svg
        class="icon icon-sm"
        enable-background="new 0 0 24 24"
        role="img"
        viewBox="0 0 24 24"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          icon
        </title>
        <path
          d="M12 1.7 9.5 9.2H1.6L8 13.9l-2.4 7.6 6.4-4.7 6.4 4.7-2.4-7.6 6.4-4.7h-7.9L12 1.7z"
        />
      </svg>
      <span
        class="visually-hidden"
      >
        Valuta 3 stelle su 5
      </span>
    </label>
    <input
      aria-describedby="star2bDescription"
      class=""
      id="star2b"
      name="ratingB"
      type="radio"
      value="2"
    />
    <label
      class="full form-label"
      for="star2b"
    >
      <svg
        class="icon icon-sm"
        enable-background="new 0 0 24 24"
        role="img"
        viewBox="0 0 24 24"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          icon
        </title>
        <path
          d="M12 1.7 9.5 9.2H1.6L8 13.9l-2.4 7.6 6.4-4.7 6.4 4.7-2.4-7.6 6.4-4.7h-7.9L12 1.7z"
        />
      </svg>
      <span
        class="visually-hidden"
      >
        Valuta 2 stelle su 5
      </span>
    </label>
    <input
      aria-describedby="star1bDescription"
      class=""
      id="star1b"
      name="ratingB"
      type="radio"
      value="1"
    />
    <label
      class="full form-label"
      for="star1b"
    >
      <svg
        class="icon icon-sm"
        enable-background="new 0 0 24 24"
        role="img"
        viewBox="0 0 24 24"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          icon
        </title>
        <path
          d="M12 1.7 9.5 9.2H1.6L8 13.9l-2.4 7.6 6.4-4.7 6.4 4.7-2.4-7.6 6.4-4.7h-7.9L12 1.7z"
        />
      </svg>
      <span
        class="visually-hidden"
      >
        Valuta 1 stelle su 5
      </span>
    </label>
  </fieldset>
</div>
`;

exports[`Stories Snapshots Documentazione/Componenti/Rating RatingReadOnly 1`] = `
<div>
  <fieldset
    class="rating rating-read-only rating rating-label mb-3"
  >
    <legend
      class="visually-hidden"
    >
      Valutazione 4 stelle su 5
    </legend>
    <input
      aria-describedby="star5cDescription"
      aria-hidden="true"
      class=""
      disabled=""
      id="star5c"
      name="ratingC"
      type="radio"
      value="5"
    />
    <label
      class="full form-label"
      for="star5c"
    >
      <svg
        class="icon icon-sm"
        enable-background="new 0 0 24 24"
        role="img"
        viewBox="0 0 24 24"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          icon
        </title>
        <path
          d="M12 1.7 9.5 9.2H1.6L8 13.9l-2.4 7.6 6.4-4.7 6.4 4.7-2.4-7.6 6.4-4.7h-7.9L12 1.7z"
        />
      </svg>
      <span
        class="visually-hidden"
      >
        Valuta 5 stelle su 5
      </span>
    </label>
    <input
      aria-describedby="star4cDescription"
      aria-hidden="true"
      checked=""
      class=""
      disabled=""
      id="star4c"
      name="ratingC"
      type="radio"
      value="4"
    />
    <label
      class="full form-label"
      for="star4c"
    >
      <svg
        class="icon icon-sm"
        enable-background="new 0 0 24 24"
        role="img"
        viewBox="0 0 24 24"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          icon
        </title>
        <path
          d="M12 1.7 9.5 9.2H1.6L8 13.9l-2.4 7.6 6.4-4.7 6.4 4.7-2.4-7.6 6.4-4.7h-7.9L12 1.7z"
        />
      </svg>
      <span
        class="visually-hidden"
      >
        Valuta 4 stelle su 5
      </span>
    </label>
    <input
      aria-describedby="star3cDescription"
      aria-hidden="true"
      class=""
      disabled=""
      id="star3c"
      name="ratingC"
      type="radio"
      value="3"
    />
    <label
      class="full form-label"
      for="star3c"
    >
      <svg
        class="icon icon-sm"
        enable-background="new 0 0 24 24"
        role="img"
        viewBox="0 0 24 24"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          icon
        </title>
        <path
          d="M12 1.7 9.5 9.2H1.6L8 13.9l-2.4 7.6 6.4-4.7 6.4 4.7-2.4-7.6 6.4-4.7h-7.9L12 1.7z"
        />
      </svg>
      <span
        class="visually-hidden"
      >
        Valuta 3 stelle su 5
      </span>
    </label>
    <input
      aria-describedby="star2cDescription"
      aria-hidden="true"
      class=""
      disabled=""
      id="star2c"
      name="ratingC"
      type="radio"
      value="2"
    />
    <label
      class="full form-label"
      for="star2c"
    >
      <svg
        class="icon icon-sm"
        enable-background="new 0 0 24 24"
        role="img"
        viewBox="0 0 24 24"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          icon
        </title>
        <path
          d="M12 1.7 9.5 9.2H1.6L8 13.9l-2.4 7.6 6.4-4.7 6.4 4.7-2.4-7.6 6.4-4.7h-7.9L12 1.7z"
        />
      </svg>
      <span
        class="visually-hidden"
      >
        Valuta 2 stelle su 5
      </span>
    </label>
    <input
      aria-describedby="star1cDescription"
      aria-hidden="true"
      class=""
      disabled=""
      id="star1c"
      name="ratingC"
      type="radio"
      value="1"
    />
    <label
      class="full form-label"
      for="star1c"
    >
      <svg
        class="icon icon-sm"
        enable-background="new 0 0 24 24"
        role="img"
        viewBox="0 0 24 24"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          icon
        </title>
        <path
          d="M12 1.7 9.5 9.2H1.6L8 13.9l-2.4 7.6 6.4-4.7 6.4 4.7-2.4-7.6 6.4-4.7h-7.9L12 1.7z"
        />
      </svg>
      <span
        class="visually-hidden"
      >
        Valuta 1 stelle su 5
      </span>
    </label>
  </fieldset>
</div>
`;
